### PR TITLE
Add retry functionality in download

### PR DESCRIPTION
**Pull request by ayan-b**

### CODE REVIEW

File: xena_gdc_etl/utils.py [hunk 1]
```diff
@@ -178,3 +181,23 @@ def reduce_json_array(j):
     else:
         reduced = j
     return reduced
+
+
+def requests_retry_session(
+    retries=5,
+    backoff_factor=0.3,
+    status_forcelist=(500, 502, 504),
```
Comment (yunhailuo) [inline xena_gdc_etl/utils.py]: Do we know if it fails on these three code? And just these three?

**Comment by ayan-b** [comment]
> Have you tried this on one project with all data types (at least all genomics)?

Yes, it retries.

File: xena_gdc_etl/utils.py (hunk 1 above)
Comment (ayan-b) [reply xena_gdc_etl/utils.py]: So, we should disable it?

**Comment by yunhailuo** [comment]
> > Have you tried this on one project with all data types (at least all genomics)?
> 
> Yes, it retries.

Interesting. It leaves trace in the log so you can tell?

File: xena_gdc_etl/utils.py (hunk 1 above)
Comment (yunhailuo) [reply xena_gdc_etl/utils.py]: What do you mean by "disable"?

**Comment by ayan-b** [comment]
Yes, like this:
```
2019-06-12 18:35:58 [WARNING]: Retrying (Retry(total=2, connect=3, read=2$
2019-06-12 18:45:47 [WARNING]: Retrying (Retry(total=2, connect=3, read=2$
2019-06-12 18:48:55 [WARNING]: No methylation27 data for cohort TCGA-DLBC.
```

**Comment by yunhailuo** [comment]
> Yes, like this:
> 
> ```
> 2019-06-12 18:35:58 [WARNING]: Retrying (Retry(total=2, connect=3, read=2$
> 2019-06-12 18:45:47 [WARNING]: Retrying (Retry(total=2, connect=3, read=2$
> 2019-06-12 18:48:55 [WARNING]: No methylation27 data for cohort TCGA-DLBC.
> ```

Oh. BTW, which log is this? Is it `etl.err`?

**Comment by ayan-b** [comment]
Yes, it is `etl.err`.

File: xena_gdc_etl/utils.py (hunk 1 above)
Comment (ayan-b) [reply xena_gdc_etl/utils.py]: I mean it will retry regardless of error code.

Comment (yunhailuo) [reply xena_gdc_etl/utils.py]: Sound too general. But I guess it doesn't hurt. Thanks.

**Pull request merged**